### ISSUE TITLE
Channel 'unread' count does not update on correctly

### ISSUE THREAD

**Issue by alexlopezit** [GetStream/stream-chat-react#312]
Looks like the is an issue updating unread counts on channels that have not been preloaded.

Setup:

- Have 3 channels #a, #b, #c
- Load only 2 channels
```
<ChannelList options={ limit: 2  } sort = { last_message_at: -1, cid: 1 } ... />
```

- Now login with 2 users (**User 1 and User 2**) on different browsers, they must be members of the mentioned channels.
- With **User 2**, type something on channel #c the reload the browser to make sure that channel is displayed
-  After reload, lets say that we can see channel #c and channel #a plus the load more button
- **User 1** is not on channel #c

Reproduce the bug
- **User 1** types something on channel #a -> (User 2 has 1 unread on channel #a)
- **User 1** types something on channel #b -> (Channel #b appears to User 2 with 1 unread)
- Now **User 1** types something else on channel #b -> (Channel #b for User 2 still has 1 unread but should be 2)

**Comment by alexlopezit** [comment]
Hi @jaapbakker88 sure, I am currently using version stream-chat-react v1.0.1

This is how the component looks in the code:

```
    const filters = {
      type: 'messaging',
      members: { $in: [USER_ID] },
      team: TEAM_ID
    }
    const options = {
      limit: 2, // The number of channels to return (max is 30)
      watch: true
    }
    const sort = {
      last_message_at: -1,
      cid: 1
    }

  <ChannelList
    filters={filters}
    options={options}
    sort={sort}
    client={client}
    EmptyStateIndicator={() => <div>You have no channels currently</div>}
    List={ChannelListMessenger}
  />
```

Here is a screen recording of what I am experiencing for better understanding https://drive.google.com/file/d/1-LLdG-HYGtJDgvi2vZXKqse6SxT8jTWZ/view?usp=sharing

**Comment by alexlopezit** [comment]
Also just noticed that when we send a `/giphy`, to channel count gets increased, but the `total_unread_count` does not, so the count does not match for this reason as well.

**Comment by jaapbakker88** [comment]
@alexlopezit just reproduced the error on my side as well. Will investigate why this is happening!

**Comment by jaapbakker88** [comment]
@alexlopezit I found the issue and fixed it on version `1.2.2`. Regarding the /giphy issue, this seems to be an issue on the backend and I've notified the backend team about it.

**Comment by alexlopezit** [comment]
Thanks for that @jaapbakker88 I checked and seems resolved.
Do you know when the `total_unread_count` in the back ends is going to be fix so it matches the counter on channels when using `/giphy`?

cc @tbarbugli
